src/test/regress/expected/single_shard_table_udfs.out

```diff
@@ -233,6 +233,243 @@ SELECT COUNT(*) = 0 FROM pg_dist_shard WHERE logicalrelid::text LIKE '%null_dist
+-- no changes because it's already balanced
+SELECT rebalance_table_shards();
+ rebalance_table_shards
+---------------------------------------------------------------------
+
+(1 row)
+
+-- same placements
+SELECT shardid, nodeport FROM pg_dist_shard_placement WHERE shardid > 1820000 ORDER BY shardid;
+ shardid | nodeport
+---------------------------------------------------------------------
+ 1820002 |    57638
+ 1820003 |    57638
+ 1820004 |    57638
+ 1820005 |    57637
+ 1820006 |    57638
+ 1820007 |    57637
+ 1820008 |    57638
+(7 rows)
+
+-- manually move 2 shard from 2 colocation groups to make the cluster unbalanced
+SELECT citus_move_shard_placement(1820005, 'localhost', :worker_1_port, 'localhost', :worker_2_port);
+ citus_move_shard_placement
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_move_shard_placement(1820007, :worker_1_node, :worker_2_node);
+ citus_move_shard_placement
+---------------------------------------------------------------------
+
+(1 row)
+
+-- all placements are located on worker 2
+SELECT shardid, nodeport FROM pg_dist_shard_placement WHERE shardid > 1820000 ORDER BY shardid;
+ shardid | nodeport
+---------------------------------------------------------------------
+ 1820002 |    57638
+ 1820003 |    57638
+ 1820004 |    57638
+ 1820005 |    57638
+ 1820006 |    57638
+ 1820007 |    57638
+ 1820008 |    57638
+(7 rows)
+
+-- move some of them to worker 1 to balance the cluster
+SELECT rebalance_table_shards();
+NOTICE:  Moving shard xxxxx from localhost:xxxxx to localhost:xxxxx ...
+NOTICE:  Moving shard xxxxx from localhost:xxxxx to localhost:xxxxx ...
+ rebalance_table_shards
+---------------------------------------------------------------------
+
+(1 row)
+
+-- the final status, balanced
+SELECT shardid, nodeport FROM pg_dist_shard_placement WHERE shardid > 1820000 ORDER BY shardid;
+ shardid | nodeport
+---------------------------------------------------------------------
+ 1820002 |    57637
+ 1820003 |    57637
+ 1820004 |    57637
+ 1820005 |    57637
+ 1820006 |    57638
+ 1820007 |    57638
+ 1820008 |    57638
+(7 rows)
+
+-- verify we didn't break any colocations
+SELECT logicalrelid, colocationid FROM pg_dist_partition WHERE logicalrelid::text LIKE '%single_shard_table_col%' ORDER BY colocationid;
+       logicalrelid        | colocationid
+---------------------------------------------------------------------
+ single_shard_table_col1_1 |       198001
+ single_shard_table_col1_2 |       198001
+ single_shard_table_col1_3 |       198001
+ single_shard_table_col2_1 |       198002
+ single_shard_table_col3_1 |       198003
+ single_shard_table_col4_1 |       198004
+ single_shard_table_col5_1 |       198005
+(7 rows)
+
+-- again, manually move 2 shard from 2 colocation groups to make the cluster unbalanced
+-- consider using citus_drain_node when the issue is fixed: https://github.com/citusdata/citus/issues/6948
```

Comment:
It's unclear to me why the mentioned issue applies here. None of these shards have replication_factor > 1. Is it because of other left over tables from earlier in the test? If so, let's just drop those before doing the drain_node.